src/core/namespace.c

```diff
@@ -1697,6 +1772,24 @@ static int mount_overlay(const MountEntry *m) {
         return 1;
 }
 
+static int mount_bpffs(const MountEntry *m) {
+        assert(m);
+
+        _cleanup_close_ int mnt_fd = fsmount(m->bpffs_fd, 0, 0);
+        int r;
+
+        safe_close(m->bpffs_fd);
+
+        if (mnt_fd < 0)
+                log_debug_errno(errno, "Failed to fsmount: %m");
```

Comment:
Missing `return` on failure. Please improve the debugging log.
First declare `int r;`, then assertions, then fsmount():
```c
static int mount_bpffs(const MountEntry *m) {
        int r;

        assert(m);
        assert(m->bpffs_fd >= 0);

        _cleanup_close_ int mnt_fd = fsmount(m->bpffs_fd, 0, 0);
        if (mnt_fd < 0)
                return log_debug_errno(errno, "Failed to fsmount bpffs: %m");
```

`m->bpffs_fd` should not be closed here. If `bpffs_finalize()` succeed, but setting mount namespace failed before this function is called, the file descriptor is leaked.